From 6e9a05cc1024a69323ecb867775078455aac28ae Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Mon, 27 Feb 2012 20:31:05 +0100
Subject: [PATCH 106/109] ide: Support I/O status

RH-Author: Luiz Capitulino <lcapitulino@redhat.com>
Message-id: <1330374668-8838-4-git-send-email-lcapitulino@redhat.com>
Patchwork-id: 37642
O-Subject: [PATCH RHEL6.3 qemu-kvm 3/6] ide: Support I/O status
Bugzilla: 797186
RH-Acked-by: Markus Armbruster <armbru@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Reviewed-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit 50fb19008fb91e3825f18b69f4adcae2db9863eb)
---
 hw/ide/core.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 hw/ide/core.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/hw/ide/core.c b/hw/ide/core.c
index e6bbdf8..8843972 100644
--- a/hw/ide/core.c
+++ b/hw/ide/core.c
@@ -479,6 +479,7 @@ static int ide_handle_rw_error(IDEState *s, int error, int op)
         s->bus->error_status = op;
         bdrv_mon_event(s->bs, BDRV_ACTION_STOP, error, is_read);
         vm_stop(RUN_STATE_IO_ERROR);
+        bdrv_iostatus_set_err(s->bs, error);
     } else {
         if (op & BM_STATUS_DMA_RETRY) {
             dma_buf_commit(s, 0);
@@ -1893,6 +1894,7 @@ int ide_init_drive(IDEState *s, BlockDriverState *bs, const char *version)
         pstrcpy(s->version, sizeof(s->version), QEMU_VERSION);
     }
     ide_reset(s);
+    bdrv_iostatus_enable(bs);
     return 0;
 }
 
-- 
1.7.7.6

